miniapp/miniapp.cpp

```diff
@@ -202,11 +198,7 @@ void banner() {
     std::cout << "  starting miniapp\n";
     std::cout << "  - " << threading::description() << " threading support\n";
     std::cout << "  - communication policy: " << std::to_string(global_policy::kind()) << " (" << global_policy::size() << ")\n";
-#ifdef NMC_HAVE_CUDA
-    std::cout << "  - gpu support: on\n";
-#else
-    std::cout << "  - gpu support: off\n";
-#endif
+    std::cout << "  - gpu support: " << (config::has_cuda? "on": "off") << "\n";
```

Comment:
This parameter is a constant, that is set to `true` if the GPU back end was enabled at compilation, `false` otherwise.
The reasoning behind the parameters (which you can find in `src/util/config.hpp`) is to remove preprocessor cruft like `#ifdef NMC_HAVE_CUDA` from most of the code. They  are `constexpr bool` values, that can be used in `if` statements, and as template parameters, etc, unllke preprocessor macros.